### PR TITLE
Splits engine into multiple parts

### LINKED ISSUE #717: Refactor engine.py
Idea is to refactor engine.py into several files:
- engine.py
- events.py
- ...

cc @justusschock

### PR DESCRIPTION
**Pull request by justusschock**
Fixes #717 

Description:

Check list:
* [x] New tests are added (if a new feature is added) [Not Necessary]
* [x] New doc strings: description and/or example code are in RST format [Not Necessary]
* [x] Documentation is updated (if required) [Not Necessary]

### CODE REVIEW

**Comment by vfdev-5** [comment]
I tried to fixed pep8 using github interface.

**Comment by justusschock** [comment]
I fixed it now (there were still some wrong function calls, which I did not see in the first place)

**Comment by vfdev-5** [comment]
@justusschock yeah I see, I'll merge #723 and then this one.

File: ignite/engine/utils.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import inspect
```
Comment (vfdev-5) [inline ignite/engine/utils.py]: @justusschock I wonder if this file should not be private? 
We do not really provide `ReproducibleBatchSampler` as lib's API...

Comment (justusschock) [reply ignite/engine/utils.py]: Well we could for sure make it private. But I thought it's fine, since we don't expose it in the `__init__.py`

**Pull request merged**